File: src/Model/Vector.java
package Model;

import java.util.List;
import java.util.ArrayList;

/**
 * Mathematically defined as a quantity with direction as well as magnitude.
 */
public class Vector {

    //instance variables
    List<Integer> vector;
    int dimensions;

    //constructor method
    public Vector(ArrayList<Integer> vector) {
        validate(vector);
        this.vector = vector;
        setDimensions();
    }

    //private update methods

    /**
     * Validation check to ensure than the given vector is not empty.
     */
    private boolean validate(ArrayList<Integer> vector) {
        if(vector.isEmpty())
            return false;
        return true;
    }

    /**
     * Returns the number of dimensions in a given vector (zero if null).
     */
    private void setDimensions() {
        int count = 0;
        for(int i = 0; i < vector.size(); i++) {
            count++;
        }
        dimensions = count;
    }

    //public update methods
    public void setVector(ArrayList<Integer> newVector) {
        validate(newVector);
        vector = newVector;
        setDimensions();
    }

    //public access methods
    public int getDimensions() {
        return dimensions;
    }

    public List<Integer> getVector() {
        return vector;
    }
}

File: src/Model/Vector.kt
package Model

/**
 * Mathematically defined as a quantity with direction as well as magnitude.
 */
class Vector(vector: ArrayList<Int?>) {
    //instance variables
    var vector: MutableList<Int?>
        private set

    //public access methods
    var dimensions: Int = 0
        private set

    //constructor method
    init {
        validate(vector)
        this.vector = vector
        setDimensions()
    }

    /** validation check to ensure than the given vector is not empty. */
    private fun validate(vector: ArrayList<Int?>): Boolean {
        return vector.isNotEmpty()
    }

    /** returns the number of dimensions in a given vector (zero if null). */
    private fun setDimensions() {
        var count = 0
        for (i in vector.indices) {
            count++
        }
        dimensions = count
    }

    fun setVector(newVector: ArrayList<Int?>) {
        validate(newVector)
        vector = newVector
        setDimensions()
    }
}
